core/src/main/scala/org/http4s/Message.scala

```diff
@@ -267,6 +268,33 @@ final class Request[F[_]](
 
   def queryString: String = uri.query.renderString
 
+  /** cURL representation of the request.
+    *
+    * Supported cURL-Parameters are: -X, -H
+    *
+    */
+  def asCurl(
+      redactHeadersWhen: CaseInsensitiveString => Boolean = Headers.SensitiveHeaders.contains)
+      : String = {
+    /**
+      * escapes characters that are used in the curl-command, such as '
+      */
+    def escapeQuotationMarks(s: String) = s.replaceAll("'", "'\\''")
```

Comment:
Close! Scala's string escaping reduces the `\\` to a `\`, and `replaceAll` escapes the second `'` with the remaining `\`.

```
scala> "'".replaceAll("'", "'\\''")
res0: String = '''
```

You need to get a `\\` all the way to the regex.  You can add a second layer of escaping:

```
scala> "'".replaceAll("'", "'\\\\''")
res2: String = '\''
```

Or use triple quotes so each backslash gets to `replaceAll`:

```
scala> "'".replaceAll("'", """'\\''""")
res1: String = '\''
```

Reply:
Ahhh. Okay, already got lost in the backslash-quote jungle. Will rework that in a few then!
Thanks for clarifying again :-)